src/main/java/nl/tudelft/jpacman/npc/ghost/Navigation.java

```diff
@@ -40,35 +40,40 @@ private Navigation() {
 	 */
 	public static List<Direction> shortestPath(Square from, Square to,
 			Unit traveller) {
-		if (from == to) {
-			return new ArrayList<>();
-		}
-
+		List<Direction> path = null;
 		List<Node> targets = new ArrayList<>();
+		Node n = new Node(null, from, null);
 		Set<Square> visited = new HashSet<>();
-		targets.add(new Node(null, from, null));
-		while (!targets.isEmpty()) {
-			Node n = targets.remove(0);
-			Square s = n.getSquare();
-			if (s == to) {
-				List<Direction> path = n.getPath();
-				return path;
-			}
-			visited.add(s);
-			addNewTargets(traveller, targets, visited, n, s);
+
+		targets.add(n);
+
+		while (from!=to && !targets.isEmpty()) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543478/issues/source?bid=3267846&fileBranchId=3281829#l50)